### PR TITLE
Featured categories

### PR DESCRIPTION
**Pull request by mohammed-sas**
### Features
1.user can click on featured categories in homepage which would redirect to explore page with selected category.
2.user can select categories in explore page

### CODE REVIEW

File: src/pages/video listing/VideoListing.jsx [hunk 1]
```diff
@@ -5,29 +5,60 @@ import VideoCard from "../../components/video card/VideoCard";
...
+             setCategoryVideos(result);
+             setCategory(type);
+           }else{
+           setCategoryVideos(response.data.videos);
+           }
+          };
       } catch (error) {
         console.log(error);
       }
     };
```
Comment (gadgetfather) [inline src/pages/video listing/VideoListing.jsx:36]: You can define this function out of useEffect

File: src/pages/video listing/VideoListing.jsx [hunk 2]
```diff
@@ -5,29 +5,60 @@ import VideoCard from "../../components/video card/VideoCard";
...
-
     fetchVideos();
   }, []);
+  const categoryHandler=(category)=>{
+    setSearchParams({"type":category});
+    setCategory(category);
+    if(category === "all"){
+      setCategoryVideos(lists);
+      return;
```
Comment (gadgetfather) [inline src/pages/video listing/VideoListing.jsx:44]: is this `return` by mistake left here?

**Review by gadgetfather — APPROVED** [review]
- Functionality is working perfectly
- Code is clean
- All your function names are very descriptive 

Great work @mohammed-sas

File: src/pages/video listing/VideoListing.jsx (hunk 2 above)
Comment (mohammed-sas) [reply src/pages/video listing/VideoListing.jsx:44]: Nope

**Review by AtulPant2704 — APPROVED** [review]
The functionality works fine and code is well structured, good job.

**Pull request merged**